fdbserver/workloads/WriteTagThrottling.actor.cpp

```diff
@@ -0,0 +1,229 @@
+/*
+ * WriteThrottling.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+//
+// Created by sfc-gh-xwang on 6/29/20.
+//
+#include "fdbserver/TesterInterface.actor.h"
+#include "fdbserver/workloads/workloads.actor.h"
+#include "fdbserver/WorkerInterface.actor.h"
+#include "fdbclient/NativeAPI.actor.h"
+#include "fdbclient/TagThrottle.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+// workload description:
+// This workload aims to test whether we can throttling some bad clients that doing penetrating write on write hot-spot
+// range. There are several good clientActor just randomly do read and write ops in transaction. Also, some bad
+// clientActor has high probability to read and write a particular hot-spot. If the tag-based throttling works right on
+// write-heavy tags, it will only limit the bad clientActor without influence other normal actors. This workload also
+// output TPS and latency of read/set/clear operations to do eyeball check. We want this new feature would not cause
+// more load to the cluster (Maybe some visualization tools is needed to show the trend of metrics) Note: this workload
+// may not be proper for simulation
```

Comment:
at the minimum, this note should be highlighted at the top of the description.

It will be very valuable to make this test suitable for simulation. 
For example, if we make the extreme heavy traffic to 1-2 tags and light traffic to other tags, we may trigger the throttling. In the client side, the bad client (that has lot of traffics) can measure the rate it generates, monitor the feedback from the tagThrottling feature, and cross-check if these two metrics are consistent.

Reply:
I agree that having something that is suitable for simulation would be very useful, however:

> we may trigger the throttling

This is the problem. We can't fail on fdb not throttling I think. But we definitely should think about this a bit more...

Reply:
Now I think this workload can be used in simulation.